src/core_plugins/kibana/public/dashboard/top_nav/clone_modal.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  KuiModal,
+  KuiModalHeader,
+  KuiModalHeaderTitle,
+  KuiModalBody,
+  KuiModalBodyText,
+  KuiModalFooter,
+  KuiButton
+} from 'ui_framework/components';
+
+export class DashboardCloneModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newDashboardName: props.title
+    };
+  }
+
+  componentDidMount() {
+    this.confirmButton.focus();
+  }
+
+  cloneDashboard = () => {
+    this.props.onClone(this.state.newDashboardName);
+  };
+
+  onInputChange = (event) => {
+    this.setState({ newDashboardName: event.target.value });
+  };
+
+  onKeyDown = (event) => {
+    if (event.keyCode === 27) {
```

Comment:
We have it for all the confirm modals currently.  Or at least, it should be working for our confirm modals....  if it's not it's a bug.